generate.sh

```diff
@@ -96,66 +106,58 @@ EOF
 for version in ${VERSIONS}; do
     # copy targets
     rules="$COPY_RULES"
-    core="cp \$< \$@ ; \\"
-    message="Copying \"\$@\""
+    core="cp \$< \$@"
     creator="copy"
     parse_rules
     COPY_TARGETS+="$targets"
 
 
     # distgen targets
     rules="$DISTGEN_RULES"
-    core="${DG} --multispec specs/multispec.yml \\
+    # TODO: Drop terrible hack that we use fixed --distro here!
+    core="\$(DG) --multispec specs/multispec.yml \\
 	--template \"\$<\" --distro centos-7-x86_64.yaml \\
-	--multispec-selector version=\"$version\" --output \"\$@\" ; \\"
-    message="Generating \"\$@\" using distgen"
+	--multispec-selector version=\"$version\" --output \"\$@\""
     creator="distgen"
     parse_rules
     DISTGEN_TARGETS+="$targets"
 
 
     rules=$SYMLINK_RULES
-    core="ln -fs __link_target__ \$@ ; \\"
-    message="Creating symlink \"\$@\""
+    core="ln -fs __link_target__ \$@"
     creator="link"
     parse_rules
     SYMLINK_TARGETS+="$targets"
 done
 
 while read -r combination; do
-    version=${combination##*=}
-    DG_CONF=$(echo $combination | cut -d' ' -f2)
+    # line looks like: --distro rhel-7-x86_64.yaml --multispec-selector version=9.4
+    eval 'set -- $combination'
+    case $4 in
+        version=*) version=${4##*=} ;;
+        *) die "version not found"
+    esac
+    case $2 in
+        *x86_64*) DG_CONF=$2 ;;
```

Comment:
I wonder if it would be worth it to future proof this for additional architectures